docs/handbook/04_streams.md

```diff
@@ -127,13 +127,31 @@ Content-Type: text/vnd.turbo-stream.html; charset=utf-8
 
 This `messages/message` template partial can then also be used to re-render the message following an edit/update operation. Or to supply new messages created by other users over a WebSocket or a SSE connection. Being able to reuse the same templates across the whole spectrum of use is incredibly powerful, and key to reducing the amount of work it takes to create these modern, fast applications.
 
-
 ## Progressively Enhance When Necessary
 
 It's good practice to start your interaction design without Turbo Streams. Make the entire application work as it would if Turbo Streams were not available, then layer them on as a level-up. This means you won't come to rely on the updates for flows that need to work in native applications or elsewhere without them.
 
 The same is especially true for WebSocket updates. On poor connections, or if there are server issues, your WebSocket may well get disconnected. If the application is designed to work without it, it'll be more resilient.
 
+If a Turbo Drive visit needs to integrate with Turbo Streams, you can overwrite its [Accept][] header within a `turbo:before-fetch-request` event listener:
+
+```js
+let specialCaseRequest = false
+
+addEventListener("turbo:click", ({ target }) => {
+  specialCaseRequest = target.hasAttribute("data-special-request")
+})
+
+addEventListener("turbo:before-fetch-request", ({ detail }) => {
+  if (specialCaseRequest) {
+    detail.fetchOptions.headers.Accept = "text/vnd.turbo-stream.html"
```

Comment:
should we export `"text/vnd.turbo-stream.html"` as a const so that users dont need to hardcode it here?

Reply:
@ghiculescu @seanpdoyle A slight variation of this controller without `targets`. Check this out:

```javascript
import { Controller } from "stimulus"

export default class extends Controller {
  // Actions

  inject() {
    addEventListener("turbo:before-fetch-request", this.injectHeaders, { once: true })
  }

  // Private

  injectHeaders({ detail }) {
    const { headers } = detail.fetchOptions || {}
    if (headers) {
      headers.Accept = ["text/vnd.turbo-stream.html", headers.Accept].join(", ")
    }
  }
}
```

Now you can just use this:

```html
<body data-controller="turbo-streams">
  <a href="/foo" data-action="turbo:click->turbo-streams#inject">Load content via Turbo Stream</a>
</body>
```

Reply:
👍 good idea!